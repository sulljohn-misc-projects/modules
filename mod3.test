#!/bin/bash

# suppress pushd and popd output                                               
pushd () {                                                                     
    command pushd "$@" > /dev/null                                             
}                                                                            
popd () {                                                                     
		command popd "$@" > /dev/null                                               
}

Y=1
pushd .
while [ ${Y} -le 3 ] ; do
		X=1
		if [ ${Y} -eq 1 ] ; then
				while [ ${X} -le 10 ] ; do
						runtest.sh "list_test ${X}"
						valgrind --leak-check=yes --log-fd=1 -q "./list_test ${X}" | grep -q "in use at exit: 0 bytes in 0 blocks"
						valgrind.sh "echo $?"
						let X=X+1
				done
		elif [ ${Y} -eq 2 ] ; then
				while [ ${X} -le 13 ] ; do
            runtest.sh "queue_test ${X}"
						valgrind --leak-check=yes --log-fd=1 -q "./queue_test ${X}" | grep -q "in use at exit: 0 bytes in 0 blocks"
						valgrind.sh "echo $?"
            let X=X+1
        done
		else
				while [ ${X} -le 10 ] ; do
						runtest.sh "hash_test ${X}"
						valgrind --leak-check=yes --log-fd=1 -q "./hash_test ${X}" | grep -q "in use at exit: 0 bytes in 0 blocks"
            valgrind.sh "echo $?"
						let X=X+1
				done
		fi
		let Y=Y+1
done
popd
