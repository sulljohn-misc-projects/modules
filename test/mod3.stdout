Success  : list_test 1
No memory leaks detected
Success  : list_test 2
No memory leaks detected
Success  : list_test 3
No memory leaks detected
Success  : list_test 4
No memory leaks detected
Success  : list_test 5
No memory leaks detected
Success  : list_test 6
No memory leaks detected
Success  : list_test 7
No memory leaks detected
Success  : list_test 8
No memory leaks detected
Success  : list_test 9
No memory leaks detected
Success  : list_test 10
No memory leaks detected
Success  : queue_test 1
No memory leaks detected
Success  : queue_test 2
No memory leaks detected
Success  : queue_test 3
No memory leaks detected
Success  : queue_test 4
No memory leaks detected
Success  : queue_test 5
No memory leaks detected
Success  : queue_test 6
No memory leaks detected
Success  : queue_test 7
No memory leaks detected
Success  : queue_test 8
No memory leaks detected
Success  : queue_test 9
No memory leaks detected
Success  : queue_test 10
No memory leaks detected
Success  : queue_test 11
No memory leaks detected
Success  : queue_test 12
No memory leaks detected
Success  : queue_test 13
No memory leaks detected
Success  : hash_test 1
No memory leaks detected
Success  : hash_test 2
No memory leaks detected
Success  : hash_test 3
No memory leaks detected
Success  : hash_test 4
No memory leaks detected
Success  : hash_test 5
No memory leaks detected
Success  : hash_test 6
No memory leaks detected
Success  : hash_test 7
No memory leaks detected
Success  : hash_test 8
No memory leaks detected
Success  : hash_test 9
No memory leaks detected
Success  : hash_test 10
No memory leaks detected
